Sources/Turf/Geometries/Polygon.swift

```diff
@@ -297,4 +297,55 @@ extension Polygon {
             ring[2].longitude == ring[0].longitude
         )
     }
-}
+
+
+    /// Calculates the centroid using the mean of all vertices.
+    /// This lessens the effect of small islands and artifacts when calculating the centroid of a set of polygons.
+    public func centroid() -> LocationCoordinate2D? {
+        let coordinates = outerRing.coordinates.dropLast()
+        let summed = coordinates
+            .reduce(into: LocationCoordinate2D(latitude: 0, longitude: 0)) { acc, next in
```

Comment:
The JavaScript implementation [relies on the feature’s bounding box](https://github.com/Turfjs/turf/blob/ae066ae320a5e6a95960f2ed608d79d52b7b62d8/packages/turf-center/index.ts#L40). I’d imagine that this implementation is a bit more efficient than the one in `BoundingBox(from:)`. Perhaps we could replace it outright?

Reply:
They are different things. The JS implementation that you linked is for `center`. I haven't ported that one, but rather `centroid` (as `centerOfMass` depends on that). Both have different advantages and disadvantages. The bounding boxes `center` is susceptible to outliers, while `centroid` is susceptive to inbalanced polygons.

See for example:
- center: https://github.com/Turfjs/turf/blob/master/packages/turf-center/test/out/imbalanced-polygon.geojson
- centroid: https://github.com/Turfjs/turf/blob/master/packages/turf-centroid/test/out/imbalanced-polygon.geojson

I can add `center`, too, as it's pretty simple.